src/Listeners/UpdateSyncedResource.php

```diff
@@ -112,11 +113,24 @@ protected function updateResourceInTenantDatabases($tenants, $event, $syncedAttr
                     $localModel->update($syncedAttributes);
                 } else {
                     // When creating, we use all columns, not just the synced ones.
-                    $localModel = $localModelClass::create($eventModel->getAttributes());
+                    $localModel = $localModelClass::create($this->getResourceCreationAttributes($eventModel));
                 }
 
                 event(new SyncedResourceChangedInForeignDatabase($localModel, $tenant));
             });
         });
     }
+
+    protected function getResourceCreationAttributes(Syncable $model): array
+    {
+        $attributes = $model->getAttributes();
+
+        if ($model->getResourceCreationAttributes()) {
+            // If function returned array is key-value, We assume default values are provided
+            // if array is plain values, fetch attributes from model
+            $attributes = Arr::isAssoc($model->getResourceCreationAttributes()) ? $model->getResourceCreationAttributes() : $model->only($model->getResourceCreationAttributes());
```

Comment:
If it's an associative array, you should *still* be able to copy values from the model. You don't want to use *just* the default values to create the model.

So in the implementation, I think they should be merged into the model's synced attributes?

Reply:
And before making any logical changes, use TDD. Update the tests to assert what we want to see, make them fail, and then modify the code to make them pass.

It's a semi complex feature and we don't want the tests to pass accidentally, so we want to see them fail first and then pass when the implementation is fixed.

Reply:
> You don't want to use just the default values to create the model.

If I understand it correctly, You are saying even if the developer specified the default value of a column, and the same column exists in the sync attributes, I should copy the value from the model instead of using the default value.

If I'm correct, What if the developer doesn't want to use the column value and wants to provide the default value always for some columns even if they exist in the sync attribute?

Reply:
> You are saying even if the developer specified the default value of a column, and the same column exists in the sync attributes

The point of defaults is that they're used when we don't have any actual value to use.

However this is not what I meant in my comment. I was talking about the **other** attributes. The code here looks like it will use ONLY the defaults. But we want to use *other columns* as well. The point of the defaults here is to fill the columns that we don't have values for. But we still want to use the columns that we do have values for.

So I'm talking about different columns, not overriding values for the same column.

Reply:
> So I'm talking about different columns, not overriding values for the same column.

I see, merging them makes sense.

Reply:
I've made the changes. As you said, I did TDD. Also, the functions are almost rewritten, and the code looks clean now.

Reply:
Resolving this review since implementation got changed and requested changes are implemented.